lti_consumer/plugin/compat.py

```diff
@@ -94,3 +94,43 @@ def publish_grade(block, user, score, possible, only_if_higher=False, score_dele
         score_deleted=score_deleted,
         grader_response=comment,
     )
+
+
+def get_user_enrollments(course_key):
+    """
+    Given a course key return a list of enrollments.
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from openedx.core.djangoapps.enrollments import api
+    # pylint: disable=protected-access
+    return api._data_api().get_user_enrollments(course_key)
+
+
+def get_user_profiles(users):
+    """
+    Given a list of user, returns corresponding user profiles
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from student.models import UserProfile
+    return UserProfile.objects.filter(user__in=users)
+
+
+def get_external_ids(users):
```

Comment:
@shimulch I'd prefer moving this to `edx-platform` and using `compat.py` merely as a mock isolation layer from the platform.

Do you think we can push this to the core platform (create a bulk get_or_create external ID method)?

Reply:
I think we can push a class method to ``ExternalId`` that does this bulk get_or_create. Should I open a PR for that?

Reply:
@giovannicimolin, https://github.com/edx/edx-platform/pull/25844 has been opened for upstreaming this.